script.js

```diff
@@ -1,4 +1,306 @@
+var currentGameMode = "waiting for user name";
+var userName = " ";
+var playCount = 0;
+var userWinCount = 0;
+var computerWinCount = 0;
+var drawCount = 0;
+var reverseMode = false;
+var mostRecentWinner = "";
+
+// create a function to generate Scissors Paper Stone
+var generateSPS = function () {
+  // produces a decimal between 0 and 2
+  var randomDecimal = Math.random() * 3;
+  // take off the decimal to get a number 0 - 2 ... add 1
+  var randomInteger = Math.floor(randomDecimal) + 1;
+  // assign 1, 2, and 3 to "scissors", "paper", and "stone" respectively.
+  if (randomInteger == 1) {
+    var randomItem = "scissors";
+  } else if (randomInteger == 2) {
+    randomItem = "paper";
+  } else if (randomInteger == 3) {
+    randomItem = "stone";
+  }
+  return randomItem;
+};
+
+var playSPS = function (userName, userGuess) {
+  var message = "";
+  // input validation
+  if (isNaN(Number(userGuess)) == false) {
```

Comment:
I like the application of what's learnt in class. Great work anticipating the possibility that the user might not enter the values we want them to enter